### PR TITLE
Andrea jaqueline

### PR DESCRIPTION
**Pull request by chilimatic**
before it's to late ;)

### CODE REVIEW

File: app/lib/transformer/string/html/HashTransformer.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    /**
+     * DRY
+     */
+    use NoOptionsTransformerTrait;
+
+    /**
+     * pattern for search and replace
+     */
+    const PATTERN = '/(^|\s)#(\w*[a-zA-Z0-9öäü_-]+\w*)/';
+    const REPLACE_PATTEN = '\1<a href="http://www.dasmerkendienie.com/hash/\2">#\2</a>';
```
Comment (andreas83) [inline app/lib/transformer/string/html/HashTransformer.php]: hostname should be part of config

File: app/lib/transformer/string/html/LinkTransformer.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ *
+ * @author j
+ * Date: 10/7/15
+ * Time: 8:35 PM
+ *
+ * File: LinkTransformer.php
+ */
+class RedtubeTransformer implements IStringTransformer
```
Comment (andreas83) [inline app/lib/transformer/string/html/LinkTransformer.php]: class name is RedtubeTransformer while Filename is LinkTransformer ?

**Comment by andreas83** [comment]
just as a quick feedback:
i highly appreciate the idea of having a request and response object, good job!
also the gulp stuff looks pretty promising, we should add the jsx transformation for react too...

File: app/lib/transformer/string/html/LinkTransformer.php (hunk 2 above)
Comment (chilimatic) [inline app/lib/transformer/string/html/LinkTransformer.php]: thx.

File: app/lib/transformer/string/html/HashTransformer.php (hunk 1 above)
Comment (chilimatic) [inline app/lib/transformer/string/html/HashTransformer.php]: thx, it wasn't there at the time. fixed

**Comment by andreas83** [comment]
i removed the jquery, bootstrap and highlight js part from composer.json 
in favor of bower...  
at moment the css and js stuff is included in footer.php, 
i would love to see a better approach, like require.js oder webpack. any recommendations ?

**Comment by chilimatic** [comment]
i use require.js in my angular projects, but for my cycleJS experiments I have webpack which is still kinda confusing. And if I remember correctly one of the guys @ the meetup mentioned that if you use webpack you have to use it for everything. 

I do know requireJS so I can help you there. I personally (at least now) prefer the bundle approach instead of the lazy loading. The lazy loading seams cool but in the end the HTTP requests add up and mapping them in JS can create the same issues as just compiling / concating them to packages for specific areas. 

But maybe there are benefits, I'd have to study them. Atm I would use require JS and create "area"-bundles for example "frontend.bundle.min.js".

**Comment by andreas83** [comment]
I moved all frontend plugins (jquery, bootstrap, autocomplert) from composer to bower. 

Todo:
- [x] gulp file (react task jsx transformation)
- [x] reduce http calls - too many js css loaded via footer.php (maybe we can uglify it or/and use require.js)
- [x] cleanup (please remove all unused node dependencies)
- [x] testing

Right now it looks most of the stuff works, but since the react taks is broken iam not able todo further test..

**Comment by chilimatic** [comment]
and which one are the unused node dependencies ? :D I can't read your mind ! :)

**Comment by andreas83** [comment]
i thought you know which dependencies are required since you wrote the package.json. 
i guess the sass part is not needed anymore?!
of cause we could move the sass parsing from php 2 gulp too.

{
  "name": "dmdn",
  "version": "0.0.1",
  "description": "dmdn socket IO chat",
  "dependencies": {
    "bower": "_",
    "sass": "_",
    "gulp": "_",
    "gulp-sass": "_",
    "gulp-minify-css": "_",
    "gulp-jshint": "_",
    "gulp-concat": "_",
    "gulp-uglify": "_",
    "gulp-livereload": "_",
    "gulp-cache" : "_",
    "gulp-notify" : "_",
    "gulp-rename" : "_",
    "gulp-babel" : "_",
    "del" : "_",
    "react-tools": "*"
  },
  "scripts": {
    "start" : ""
  }
}

**Comment by chilimatic** [comment]
ahh i see, bower is still useful for specific packages like jquery and such, 
The rest is really just gulp, I will try to fix the gulpfile, unify the "build" and maybe explain it to you tonight for 5 minutes why I do what and how and you can take it over and change the things you don't like, or tell me what you want in another way.

**Comment by andreas83** [comment]
sounds good, lets have a quick chat tonight !

**Comment by andreas83** [comment]
looks like i do not get my gulp task "react" working properly. to be more specific, 
the generated code does not work like the generated code from cli tool.

**cli works fine**
babel --presets react -w Author.jsx Comments.jsx Stream.jsx Likes.jsx SearchBox.jsx Chat.jsx ShareBox.jsx Notifications.jsx InitStream.jsx --watch --out-file react.js

**gulp task**
gulp.task('react', function() {
    return gulp.src('public/js/jsx/*.jsx')
        .pipe(babel({"presets": ["react"]}))
        .pipe(concat('react.js'))
        .pipe(gulp.dest('public/js/'))
});

any clue whats wrong ?

**Comment by andreas83** [comment]
found the soloution (its the order)

**Comment by andreas83** [comment]
i reduced the http request by moved every js into one big app.js 

what do you think ?

**Comment by chilimatic** [comment]
sorry I was working till now, I would do the same thing maybe group it into specific bundles.
Btw a guy from Mobfox mentioned that you (everyone) should look out for the re-rendering in react because most developers forget to check if the component needs to be re-rendered. 

Do you stilll need my help ? otherwise I would just leave it to your capable hands

**Comment by andreas83** [comment]
re rendering ? 
for now the gulp task works for me, every javascript  (jquery, react, bootstrap etc) is now one uglified app.js. 

Pro
- we do not depend on any external source (will work on the intranet)

Con 
- Not sure if one big js file is better than 10 small ones.

thx for your help, i highly appreciate your work (in special the request, response part, as well the gulp stuff) 

next task for me will be testing and merge the whole thing, also i like to focus on documentation of the development stack. 
Lets talks about phing and unit test next time!

**Comment by chilimatic** [comment]
1 big is less blocking than 10 small ones. So For me it always wins, bigger companies like google deliver 1 basic small inline package and load the bigger scripts afterwards. but that's a different topic :)

File: composer.json [hunk 3]
```diff
@@ -7,11 +7,10 @@
         "altorouter/altorouter"     : "^1.1",
         "leafo/scssphp"             : "^0.6.3",
         "phpmailer/phpmailer"       : "^5.2",
-        "league/oauth2-client": "^1.3",
-        "league/oauth2-facebook": "~1.0",
-        "league/oauth2-google": "^1.0",
-        "textalk/websocket": "^1.2"
-
+        "league/oauth2-client"      : "^1.3",
```
Comment (andreas83) [inline composer.json:9]: :heart: :heart: :heart:

**Pull request merged**